src/containers/viewAllArticles.js

```diff
@@ -7,24 +7,28 @@ import ViewCard from '../components/articles/viewCard';
 import {fetchAllArticles} from '../actions/articleActions/articleActions';
 import {viewAllArticleActionCreator} from '../actions/articleActions/articleActionCreators';
 import Pagination from './pagination';
+import FilterArticles from './filterArticles';
 
 class ViewAllArticles extends React.Component {
 
 	constructor(props) {
 		super(props);
+
+		const articles = this.props.articles || this.props.article;
+
 		this.state = {
-			articles: this.props.articles,
 			pageCount: 0,
-			page: 1,
-			limit: 20,
-			offset: 0
+			page: 1, limit: 20, offset: 0,
+			tag: '', title: '', author: '',
+			articles
 		};
 	}
 
 	componentWillMount() {
 		const { articles } = this.state;
+		const { results, count } = articles;
```

Comment:
for this, we can have one destructuring assignment, that is
`const {results, count} = this.state.articles;`